client/client.go

```diff
@@ -449,39 +524,205 @@ func GetTest(ctx context.Context, c shared.Config) (err error) {
+		}
+	}
+
+	if c.PrintFull {
+		printDataPointHeaders(dps[0].Type)
+		for i := range dps {
+			dp := dps[i]
+			sp1 := strings.Split(dp.Local, ":")
+			sp2 := strings.Split(sp1[0], ".")
+			s1 := lipgloss.NewStyle().Background(lipgloss.Color(getHex(sp2[len(sp2)-1])))
+			printTableRow(s1, &dp, dp.Type)
 		}
+	}
 
-		return nil
+	if c.PrintErrors {
+		for i := range errors {
+			PrintTError(errors[i])
+		}
 	}
 
-	printDataPointHeaders(responseDPS[0].Type)
-	for i := range responseDPS {
-		dp := responseDPS[i]
-		sp1 := strings.Split(dp.Local, ":")
-		sp2 := strings.Split(sp1[0], ".")
-		s1 := lipgloss.NewStyle().Background(lipgloss.Color(getHex(sp2[len(sp2)-1])))
-		printTableRow(s1, &dp, dp.Type)
+	dps10 := math.Ceil((float64(len(dps)) / 100) * 10)
+	dps90 := math.Floor((float64(len(dps)) / 100) * 90)
+
+	slices.SortFunc(dps, func(a shared.DP, b shared.DP) int {
+		if a.RMSH < b.RMSH {
+			return -1
+		} else {
+			return 1
+		}
+	})
+
+	dps10s := make([]shared.DP, 0)
+	dps50s := make([]shared.DP, 0)
+	dps90s := make([]shared.DP, 0)
+
+	// total, sum, low, mean, high
+	dps10stats := []int64{0, 0, math.MaxInt64, 0, 0}
+	dps50stats := []int64{0, 0, math.MaxInt64, 0, 0}
+	dps90stats := []int64{0, 0, math.MaxInt64, 0, 0}
+
+	for i := range dps {
+		if i <= int(dps10) {
+			dps10s = append(dps10s, dps[i])
+			updateBracketStats(dps10stats, dps[i])
+		} else if i >= int(dps90) {
+			dps90s = append(dps90s, dps[i])
+			updateBracketStats(dps90stats, dps[i])
+		} else {
+			dps50s = append(dps50s, dps[i])
+			updateBracketStats(dps50stats, dps[i])
+		}
 	}
 
-	for i := range responseERR {
-		PrintTError(responseERR[i])
+	fmt.Println("")
+	fmt.Println("")
+	fmt.Println("")
+
+	fmt.Println(" First 10% of data points")
+	printBracker(dps10stats, SuccessStyle)
+	fmt.Println("")
+	fmt.Println(" Between 10% and 90%")
+	printBracker(dps50stats, WarningStyle)
+	fmt.Println("")
+	fmt.Println(" Last 10% of data points")
+	printBracker(dps90stats, ErrorStyle)
+	fmt.Println("")
+	return nil
+}
+
+func printBracker(b []int64, style lipgloss.Style) {
+	fmt.Println(style.Render(
+		fmt.Sprintf(" Total %d | Low %d | Avg %d | High %d | Microseconds ",
+			b[0],
+			b[2],
+			b[3],
+			b[4],
+		),
+	))
+}
+
+func updateBracketStats(b []int64, dp shared.DP) {
+	b[0]++
+	b[1] += dp.RMSH
+	if dp.RMSH < b[2] {
+		b[2] = dp.RMSH
+	}
+	b[3] = b[1] / b[0]
+	if dp.RMSH > b[4] {
+		b[4] = dp.RMSH
+	}
+}
+
+func MakeCSV(ctx context.Context, c shared.Config) (err error) {
+	byteValue, err := os.ReadFile(c.File)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.Create(c.File + ".csv")
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	fb := bytes.NewBuffer(byteValue)
+	scanner := bufio.NewScanner(fb)
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+	if err := writer.Write(getStructFields(new(shared.DP))); err != nil {
+		return err
+	}
+
+	for scanner.Scan() {
+		b := scanner.Bytes()
+		if bytes.Contains(b, []byte("Error")) {
```

Comment:
Isn't there a more reliable check for this?

Reply:
@zveinn ^^

Reply:
Will see if I can't make this more reliable

Reply:
Make a byte (as string) prefix to every line.